.github/workflows/lint.yaml

```diff
@@ -0,0 +1,27 @@
+name: Lint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - main
+  pull_request:
+jobs:
+  lint:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v4
+        with:
+          go-version: "1.22.3"
```

Comment:
I'd suggest using the go.mod to extract the version. Example here: https://github.com/invopop/transform/blob/main/.github/workflows/lint.yaml#L15-L22